### PR TITLE
Give a clear error when running an agent on a finished sweep

### PR DESCRIPTION
**Pull request by vwrj**
Fixes WB-9492

Description
-----------
If you currently run an agent on a finished sweep, the following error comes up: "Error calling W&B API: Sweep {name} is not running". This is vague so this PR adds a clearer error message. 

Testing
-------
How was this PR tested?

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -1702,6 +1702,15 @@ def no_retry_4xx(e):
...
+            message = body["errors"][0]["message"]
+            if (
+                e.response.status_code == 400
+                and "Sweep" in message
+                and "is not running" in message
+            ):
+                wandb.termerror(
+                    "The sweep has finished -- all hyperparameter combinations have been exhausted."
+                )
             raise UsageError(body["errors"][0]["message"])
```
Comment (dmitryduev) [inline wandb/sdk/internal/internal_api.py:1714]:
```suggestion
            raise UsageError(message)
```

File: wandb/sdk/internal/internal_api.py [hunk 2]
```diff
@@ -1702,6 +1702,15 @@ def no_retry_4xx(e):
             ):
                 return True
             body = json.loads(e.response.content)
+            message = body["errors"][0]["message"]
```
Comment (dmitryduev) [inline wandb/sdk/internal/internal_api.py:1705]: it's always a str, right?

**Review by dmitryduev — APPROVED** [review]

File: wandb/sdk/internal/internal_api.py [hunk 3]
```diff
@@ -1702,7 +1702,16 @@ def no_retry_4xx(e):
             ):
                 return True
             body = json.loads(e.response.content)
-            raise UsageError(body["errors"][0]["message"])
+            message = body["errors"][0]["message"]
+            if (
+                e.response.status_code == 400
+                and "Sweep" in message
```
Comment (kptkin) [inline wandb/sdk/internal/internal_api.py:1708]: i'm not loving the specific workaround of using the string content to figure out this case. 
Is there a better way to get this indication? Or maybe just change the error message?

Comment (dmitryduev) [reply wandb/sdk/internal/internal_api.py:1708]: I actually agree with @kptkin, @vwrj. Is there a better way to do this?

**Comment by dmitryduev** [comment]
Hey @vwrj, any updates on this?

File: wandb/sdk/internal/internal_api.py [hunk 4]
```diff
@@ -1931,7 +1931,16 @@ def no_retry_4xx(e: Exception) -> bool:
...
+                and "is not running" in message
+            ):
+                wandb.termerror(
+                    "The sweep has finished -- all hyperparameter combinations have been exhausted."
+                )
+            raise UsageError(message)
```
Comment (dmitryduev) [inline wandb/sdk/internal/internal_api.py:1944]: Mind adding a test for this please?

**Review by dmitryduev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**